### PR TITLE
Vendor in latest containers/storage

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by mtrmac** [comment]
@rhatdan The failing tests say to run `go mod tidy`, and that’s sufficient.

**Comment by rhatdan** [comment]
That is blowing up on my system.

I tried that and I am getting:
```
 $ go mod tidy
go: downloading github.com/containers/storage v1.44.1-0.20221209084436-73d739442168
go: downloading github.com/chzyer/logex v1.2.1
go: finding module for package gotest.tools/assert/cmp
go: finding module for package gotest.tools/assert
github.com/containers/image/v5/storage imports
	github.com/containers/storage imports
	github.com/containers/storage/types tested by
	github.com/containers/storage/types.test imports
	gotest.tools/assert: package gotest.tools/assert provided by gotest.tools at latest version v1.4.0 but not at required version v2.2.0+incompatible
github.com/containers/image/v5/storage imports
	github.com/containers/storage/pkg/chunked imports
	github.com/containers/storage/drivers/copy tested by
	github.com/containers/storage/drivers/copy.test imports
	gotest.tools/assert/cmp: package gotest.tools/assert/cmp provided by gotest.tools at latest version v1.4.0 but not at required version v2.2.0+incompatible
github.com/containers/image/v5/pkg/shortnames imports
	github.com/manifoldco/promptui imports
	github.com/chzyer/readline tested by
	github.com/chzyer/readline.test imports
	github.com/chzyer/test imports
	github.com/chzyer/logex: reading github.com/chzyer/logex/go.mod at revision v1.2.1: git ls-remote -q origin in /home/dwalsh/go/pkg/mod/cache/vcs/18c99d76a024736474e5839601b6c86a9d1c934fa55c5fadc3d4d0dba0864d89: exit status 128:
	fatal: 'origin' does not appear to be a git repository
	fatal: Could not read from remote repository.
	
	Please make sure you have the correct access rights
	and the repository exists.
```

**Comment by mtrmac** [comment]
I don’t know, it seems to work fine for me, and then build successfully, with `go version go1.19.4 darwin/amd64`.

**Comment by rhatdan** [comment]
A little further but now I am getting:

 $ go mod tidy
go: finding module for package gotest.tools/assert/cmp
go: finding module for package gotest.tools/assert
github.com/containers/image/v5/storage imports
	github.com/containers/storage imports
	github.com/containers/storage/types tested by
	github.com/containers/storage/types.test imports
	gotest.tools/assert: package gotest.tools/assert provided by gotest.tools at latest version v1.4.0 but not at required version v2.2.0+incompatible
github.com/containers/image/v5/storage imports
	github.com/containers/storage/pkg/chunked imports
	github.com/containers/storage/drivers/copy tested by
	github.com/containers/storage/drivers/copy.test imports
	gotest.tools/assert/cmp: package gotest.tools/assert/cmp provided by gotest.tools at latest version v1.4.0 but not at required version v2.2.0+incompatible

**Comment by rhatdan** [comment]
Can you just take this over?

**Comment by cgwalters** [comment]
> 	github.com/chzyer/logex: reading github.com/chzyer/logex/go.mod at revision v1.2.1: git ls-remote -q origin in /home/dwalsh/go/pkg/mod/cache/vcs/18c99d76a024736474e5839601b6c86a9d1c934fa55c5fadc3d4d0dba0864d89: exit status 128

Could have been a transient failure.  I often use `export GOPROXY=https://proxy.golang.org` to avoid hitting a ton of github (or worse, mercurial) repositories.

Pulling your branch and `go mod tidy` gives me

```
diff --git a/go.sum b/go.sum
index 89437402..ddf7ab0d 100644
--- a/go.sum
+++ b/go.sum
@@ -237,10 +237,6 @@ github.com/containers/ocicrypt v1.1.0/go.mod h1:b8AOe0YR67uU8OqfVNcznfFpAzu3rdgU
 github.com/containers/ocicrypt v1.1.1/go.mod h1:Dm55fwWm1YZAjYRaJ94z2mfZikIyIN4B0oB3dj3jFxY=
 github.com/containers/ocicrypt v1.1.4-0.20220428134531-566b808bdf6f h1:hffElEaoDQfREHltc2wtFPd68BqDmzW6KkEDpuSRBjs=
 github.com/containers/ocicrypt v1.1.4-0.20220428134531-566b808bdf6f/go.mod h1:xpdkbVAuaH3WzbEabUd5yDsl9SwJA5pABH85425Es2g=
-github.com/containers/storage v1.40.1-0.20220511174027-1ec5cc8c1d07 h1:ceHIeuaTvAO3gfW06ey6N0oVHjL0tIrbTNHjzOf59C0=
-github.com/containers/storage v1.40.1-0.20220511174027-1ec5cc8c1d07/go.mod h1:Pb0l5Sm/89kolX3o2KolKQ5cCHk5vPNpJrhNaLcdS5s=
-github.com/containers/storage v1.40.2 h1:GUlHaGnrs1JOEwv6YEvkQdgYXOXZdU1Angy4wgWNgF8=
-github.com/containers/storage v1.40.2/go.mod h1:zUyPC3CFIGR1OhY1CKkffxgw9+LuH76PGvVcFj38dgs=
 github.com/containers/storage v1.41.0 h1:IsoAJ1q3s/jfHB7eoiRhvTRTcuV+ywY3CkbbgKtT5Bo=
 github.com/containers/storage v1.41.0/go.mod h1:Pb0l5Sm/89kolX3o2KolKQ5cCHk5vPNpJrhNaLcdS5s=
 github.com/coreos/bbolt v1.3.2/go.mod h1:iRUV2dpdMOn7Bo10OQBFzIJO9kkE559Wcmn+qkEiiKk=
@@ -500,7 +496,6 @@ github.com/kisielk/gotool v1.0.0/go.mod h1:XhKaO+MFFWcvkIS/tQcRk01m1F5IRFswLeQ+o
 github.com/klauspost/compress v1.11.3/go.mod h1:aoV0uJVorq1K+umq18yTdKaF57EivdYsUV+/s2qKfXs=
 github.com/klauspost/compress v1.11.13/go.mod h1:aoV0uJVorq1K+umq18yTdKaF57EivdYsUV+/s2qKfXs=
 github.com/klauspost/compress v1.15.1/go.mod h1:/3/Vjq9QcHkK5uEr5lBEmyoZ1iFhe47etQ6QUkpK6sk=
-github.com/klauspost/compress v1.15.2/go.mod h1:PhcZ0MbTNciWF3rruxRgKxI5NkcHHrHUDtV4Yw2GlzU=
 github.com/klauspost/compress v1.15.4 h1:1kn4/7MepF/CHmYub99/nNX8az0IJjfSOU/jbnTVfqQ=
 github.com/klauspost/compress v1.15.4/go.mod h1:PhcZ0MbTNciWF3rruxRgKxI5NkcHHrHUDtV4Yw2GlzU=
 github.com/klauspost/pgzip v1.2.5 h1:qnWYvvKqedOF2ulHpMG72XQol4ILEJ8k2wwRl/Km8oE=
```

FWIW

File: Makefile [hunk 1]
```diff
@@ -85,3 +85,7 @@ lint:
 	git fetch -q "https://github.com/containers/image.git" "refs/heads/main"
 	upstream="$$(git rev-parse --verify FETCH_HEAD)" ; \
 		$(GOBIN)/git-validation -q -run DCO,short-subject,dangling-whitespace -range $$upstream..HEAD
+
+vendor-in-container:
+	podman run --privileged --rm --env HOME=/root -v `pwd`:/src -w /src golang:1.17 go mod tidy
+
```
Comment (vrothberg) [inline Makefile:91]: Empty line

File: Makefile [hunk 2]
```diff
@@ -85,3 +85,7 @@ lint:
 	git fetch -q "https://github.com/containers/image.git" "refs/heads/main"
 	upstream="$$(git rev-parse --verify FETCH_HEAD)" ; \
 		$(GOBIN)/git-validation -q -run DCO,short-subject,dangling-whitespace -range $$upstream..HEAD
+
+vendor-in-container:
+	podman run --privileged --rm --env HOME=/root -v `pwd`:/src -w /src golang:1.17 go mod tidy
```
Comment (vrothberg) [inline Makefile:90]: 1.17 is already EOL.  1.19 is the latest go.

**Review by vrothberg — APPROVED** [review]
LGTM

**Pull request merged**